examples/ellbvp_msh/ellbvp_msh_demo.cc

```diff
@@ -0,0 +1,183 @@
+  phi.setZero();
+
+  // Initialize object taking care of local computations on all cells for the
+  // source f. The source is the identity function
+  lf::uscalfe::ScalarLoadElementVectorProvider<double, decltype(mf_identity)>
+      elvec_builder(fe_space, mf_identity);
+  // Invoke assembly on cells (codim == 0)
+  AssembleVectorLocally(0, dofh, elvec_builder, phi);
+
+  if (num_neumann > 0) {
+    // Select von Neumann edges
+    auto edge_sel_neu = [&reader, physical_entity_nr_neu](
+                            const lf::mesh::Entity& edge) -> bool {
+      return reader.IsPhysicalEntity(edge, physical_entity_nr_neu);
+    };
+
+    // Add contributions of von Neumann boundary conditions
+    lf::uscalfe::ScalarLoadEdgeVectorProvider<double, decltype(mf_identity),
+                                              decltype(edge_sel_neu)>
+        elvec_builder_neu(fe_space, mf_identity, edge_sel_neu);
+    AssembleVectorLocally(1, dofh, elvec_builder_neu, phi);
+  }
+
+  if (num_dirichlet > 0) {
+    // Select Dirichlet edges
+    auto edge_sel_dir = [&reader, physical_entity_nr_dir](
+                            const lf::mesh::Entity& edge) -> bool {
+      return reader.IsPhysicalEntity(edge, physical_entity_nr_dir);
+    };
+
+    // Obtain specification for shape functions on edges
+    std::shared_ptr<const lf::uscalfe::ScalarReferenceFiniteElement<double>>
+        rsf_edge_p = fe_space->ShapeFunctionLayout(lf::base::RefEl::kSegment());
+    LF_ASSERT_MSG(rsf_edge_p != nullptr, "FE specification for edges missing");
+
+    // Fetch flags and values for degrees of freedom located on Dirichlet
+    // edges.
+    // bd_flags strictly speaking would not be necessary here since only
+    // boundary edges are flagged as 'dir' anyway. In other cases this might
+    // however be necessary.
+    auto bd_flags{lf::mesh::utils::flagEntitiesOnBoundary(fe_space->Mesh(), 1)};
+    auto ess_bdc_flags_values{lf::uscalfe::InitEssentialConditionFromFunction(
+        dofh, *rsf_edge_p,
+        [&edge_sel_dir, &bd_flags](const lf::mesh::Entity& edge) -> bool {
+          return (bd_flags(edge) && edge_sel_dir(edge));
+        },
+        mf_zero)};
+    // Eliminate Dirichlet dofs from linear system
+    lf::assemble::fix_flagged_solution_components<double>(
+        [&ess_bdc_flags_values](glb_idx_t gdof_idx) {
+          return ess_bdc_flags_values[gdof_idx];
+        },
+        A, phi);
+  }
+  // Assembly completed: Convert COO matrix into CRS format using Eigen's
+  // internal conversion routines.
+  Eigen::SparseMatrix<double> A_crs = A.makeSparse();
+
+  // Solve linear system using Eigen's sparse direct elimination
+  Eigen::SparseLU<Eigen::SparseMatrix<double>> solver;
+  solver.compute(A_crs);
+  Eigen::VectorXd sol_vec = solver.solve(phi);
+
+  // Compute Energy
+  // Matrix in triplet format holding Stiffness matrix.
+  if (N_dofs > 0) {
+    lf::assemble::COOMatrix<double> Stiffness(N_dofs, N_dofs);
```

Comment:
This can be implemented much simpler once https://github.com/craffael/lehrfempp/pull/98 is merged, I will take care of it.